### ISSUE TITLE
Breeding Stall

### ISSUE THREAD

**Issue by LucidCrux** [genbtc/AutoTrimps#53]
On my last few runs, it seems like autofight is kicking in too early, being too aggressive. I don't think it is just autofight, nor am I sure what exactly is going on. I do think it only started after I got a few level into coordinated though, which is part of why it might be autofight related.

I've got double digit millions, and it is dropping the breed rate into the tens and hundreds. At one point I noticed I had something like 2500+ warnings that it was taking too long to create a new fighting group. I'm not entirely sure this is an AT issue though, as unassigned workers don't seem to be contributing to breeder count and I think they should. So I have like 4 million unassigned workers that don't do anything? It might also not be buying nurseries for some reason, I don't have limits set, but I'm at only 18 on lvl 68. Most of the time if I let the save below run for a bit, it suddenly starts buying up a whole boat load of them.

I haven't dug into the code more, but I saw a couple custom settings for battle cutoff. This seems like it should be something not requiring custom variables though. It seems like there is a check somewhere that isn't working how it should because 300+ seconds breed time is too much no matter what.

A save file is below. Not sure how much use it will be. I turned off autofight for a bit to let it start building up before I thought to grab the save and it is at the end of the string of levels where this happens. It might have to be run over from the start. From the charts, it looks like it is probably starting between 60-70 pretty consistently. My guess is that wormhole buys are kicking it off and it doesn't recover. To be honest, it is just behaving weirdly. Sometimes I can load this and it doesn't stall at all, other times it stalls again on lvl 70, but just briefly and then corrects itself. I don't know. I'll try to catch it earlier on another run I'm starting in parallel now, but I am too busy to babysit it.

http://pastebin.com/raw/ZvArvcAJ

Sorry for the pile of text, trying to be thorough.

Edit: Notice the weird spike between 60-70 (the big one later is my void maps): http://imgur.com/a/P5NYm

**Comment by LucidCrux** [comment]
I can't reproduce this now. I didn't change any settings, I've just been portaling on that same file and the results are totally different. Instead of that big bump, the graphs continues on in a more or less similar fashion to what came before. I am baffled.

I am curious if anyone gets a similar breeding error if they portal on that file and don't use any of the helium. But since even I can't repeat it, for now I'm closing this. I still think there is something hinky going on with a check somewhere, though.

**Comment by LucidCrux** [comment]
Happened again, stuck on zone 60 for an hour now. I don't know what the deal is. I didn't see what initially started it, but it was all the way back on zone 49 this time. Here is a save and settings again:
save: http://pastebin.com/raw/Kkhv7iVZ
settings: http://pastebin.com/raw/Cax80QCw

(BTW yes, I know turning breed timer setting as low it goes is better for the challenge, I changed it when I realized. That doesn't mean AT should get stuck, though.)

Edit: I ended up turning autofight off for this challenge run. It doesn't seem to work quite right with very short breed times. I am not positive that this is how the breed stall situation gets starts, but it might be. What I had happen was this: Very short breed, very quick deaths. Things are going okay, and then a new big population increase happens. This knocks the breed timer back, so the claimed time is still ~1sec or less, but actual time is a few seconds. Autofight goes anyways, knocked the timer back more, then goes into a loop of having breed times that are 9 seconds instead of ~1sec (this is even with a setting of 0.1 breed time). Another population boom hits, timer gets knocked back even further despite it still have ~1.x seconds on the right side. Autofight continues to start fights and now it doesn't even get but a bit past half pop and it now takes even longer per group.

If this keeps going, I can see how after a while, you would end up with a large population with 30sec< timer and autofight continuing despite the actual breed time per group being several minutes.

**Comment by genbtc** [comment]
TLDR: BAF2 was not coded for you.

better autofight fights early. if it causes issues, go back to the ingame autofight. Better auto fight 2 is like 1 but with additional features for high level players so probably even worse for you and thats what you have it set to. BAF2 #4 says this: Clicks fight anyway if we are dead and have >=31 NextGroupTimer and deal with the consequences by firing genetecists afterwards.  You are supposed to be able to have geneticists to fire in order for that to work, so I will at least add a check to make sure you do. But you still shouldnt be using BAF#2, i dont see whats wrong with ingame autofight ? Or at least BAF 1.

the warpstations at z60 give so much population that it affects the breed timer. 

That save says you have 1000+ Seconds on your breed timer. Its from BAF2 #4 sending groups to fight as soon as possible but no way to correct the breed timer because you have no geneticists.

You must mean the 1000% explode on death from the challenge. That must mean you're always dead. Thats a hard daily.

**Comment by genbtc** [comment]
I put in a check to not do BAF2#4 without geneticists. and I fixed a bunch of stuff, most importantly skip prestige if unbought prestige. Refresh.
